File: app/src/main/java/com/ravimishra/newstar/MainActivity.java
package com.ravimishra.newstar;

import android.app.SearchManager;
import android.content.Context;
import android.graphics.Rect;
import android.net.ConnectivityManager;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.support.design.widget.NavigationView;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.Window;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.support.v7.widget.SearchView;
import android.widget.Toast;

import com.facebook.shimmer.ShimmerFrameLayout;

public class MainActivity extends AppCompatActivity
        implements NavigationView.OnNavigationItemSelectedListener,
        FragmentTechNews.OnFragmentInteractionListener,
        FragmentScienceNews.OnFragmentInteractionListener,
        FragmentSportsNews.OnFragmentInteractionListener,
        FragmentHeathcareNews.OnFragmentInteractionListener,
        FramentTopNews.OnFragmentInteractionListener, View.OnClickListener,
        FragmentSearchNews.OnFragmentInteractionListener,
        FragmentEntertainmentNews.OnFragmentInteractionListener

{

    RecyclerView.LayoutManager layoutManager;

    RelativeLayout errorLayout;
    TabLayout tabs;
    ViewPager viewPager;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        tabs= (TabLayout) findViewById(R.id.tabs);
        // btnRetry = findViewById(R.id.btnRetry);
        errorLayout = findViewById(R.id.error_layout);
        viewPager = (ViewPager) findViewById(R.id.viewpager);
        setupViewPager(viewPager);
        tabs.setupWithViewPager(viewPager);
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);


    }



    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        SearchManager searchManager= (SearchManager) getSystemService(Context.SEARCH_SERVICE);

      final  SearchView searchView=(SearchView) menu.findItem(R.id.action_search).getActionView();
      searchView.setSearchableInfo(searchManager.getSearchableInfo(getComponentName()));
      searchView.setQueryHint("Search News here...");
      searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
          @Override
          public boolean onQueryTextSubmit(String query) {
              if(query.length()>0){
               //   hideLayout();
                  LoadFragSearch(query);
              }
              return false;
          }

          @Override
          public boolean onQueryTextChange(String newText) {
             // hideLayout();
              //LoadFragSearch();
              return false;
          }
      });
        return true;
    }

    private void setupViewPager(ViewPager viewPager) {
        TabAdapter adapter = new TabAdapter(getSupportFragmentManager());
        adapter.addFragment(new FramentTopNews(), "Top");

        adapter.addFragment(new FragmentSportsNews(), "Sports");
        adapter.addFragment(new FragmentTechNews(), "Technology");
        adapter.addFragment(new FragmentHeathcareNews(), "Healthcare");
        adapter.addFragment(new FragmentEntertainmentNews(), "Entertainment");
        adapter.addFragment(new FragmentScienceNews(), "Science");
        viewPager.setAdapter(adapter);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        // Handle navigation view item clicks here.
        int id = item.getItemId();

        if (id == R.id.nav_camera) {
            // Handle the camera action
        } else if (id == R.id.nav_gallery) {

        } else if (id == R.id.nav_slideshow) {

        } else if (id == R.id.nav_manage) {

        } else if (id == R.id.nav_share) {

        } else if (id == R.id.nav_send) {

        }

        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }


    @Override
    public void onClick(View v) {

    }

    public void hideLayout(){
   tabs.setVisibility(View.GONE);
   viewPager.setVisibility(View.GONE);
    }

    public void showLayout(){
        tabs.setVisibility(View.VISIBLE);
        viewPager.setVisibility(View.VISIBLE);
    }
public void LoadFragSearch(String query){

    Bundle bundle = new Bundle();
    bundle.putString("query",query);
    Fragment frag = new FragmentSearchNews();
    frag.setArguments(bundle);
    FragmentManager manager = getSupportFragmentManager();
    FragmentTransaction transaction = manager.beginTransaction();
    hideLayout();
    transaction.replace(R.id.mainFrame,frag,"Search Fragment").addToBackStack(null);
    transaction.commit();
}
    @Override
    public void onBackPressed() {
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START);
        }
      else  if (getSupportFragmentManager().getBackStackEntryCount() >= 0) {
            showLayout();
            getSupportFragmentManager().popBackStack();
        } else {
            super.onBackPressed();
        }

    }
}

File: app/src/main/java/com/ravimishra/newstar/MainActivity.kt
package com.ravimishra.newstar

import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.navigation.NavigationView
import com.ravimishra.newstar.FramentTopNews
import com.ravimishra.newstar.FragmentSearchNews
import androidx.recyclerview.widget.RecyclerView
import android.widget.RelativeLayout
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import android.os.Bundle
import com.ravimishra.newstar.R
import android.app.SearchManager
import android.net.Uri
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.widget.SearchView
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener,
    FramentTopNews.OnFragmentInteractionListener, View.OnClickListener,
    FragmentSearchNews.OnFragmentInteractionListener {
    var layoutManager: RecyclerView.LayoutManager? = null
    var errorLayout: RelativeLayout? = null
    lateinit var tabs: TabLayout
    var viewPager: ViewPager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        tabs = findViewById(R.id.tabs)
        // btnRetry = findViewById(R.id.btnRetry);
        errorLayout = findViewById(R.id.error_layout)
        viewPager = findViewById(R.id.viewpager)
        setupViewPager(viewPager)
        tabs.setupWithViewPager(viewPager)
        //        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
//        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
//                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
//        drawer.addDrawerListener(toggle);
//        toggle.syncState();
//
//        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
//        navigationView.setNavigationItemSelectedListener(this);
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        val searchManager = getSystemService(SEARCH_SERVICE) as SearchManager
        val searchView = menu.findItem(R.id.action_search).actionView as SearchView
        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName))
        searchView.queryHint = "Search News here..."
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                if (query.length > 0) {
                    //   hideLayout();
                    LoadFragSearch(query)
                }
                return false
            }

            override fun onQueryTextChange(newText: String): Boolean {
                // hideLayout();
                //LoadFragSearch();
                return false
            }
        })
        return true
    }

    private fun setupViewPager(viewPager: ViewPager?) {
        val adapter = TabAdapter(supportFragmentManager)
        adapter.addFragment(FramentTopNews.newInstance("",true), "Top")
        adapter.addFragment(FramentTopNews.newInstance("Sports"), "Sports")
        adapter.addFragment(FramentTopNews.newInstance("Technology"), "Technology")
        adapter.addFragment(FramentTopNews.newInstance("Healthcare"), "Healthcare")
        adapter.addFragment(FramentTopNews.newInstance("Entertainment"), "Entertainment")
        adapter.addFragment(FramentTopNews.newInstance("Science"), "Science")
        viewPager!!.adapter = adapter
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        val id = item.itemId
        if (id == R.id.nav_camera) {
            // Handle the camera action
        } else if (id == R.id.nav_gallery) {
        } else if (id == R.id.nav_slideshow) {
        } else if (id == R.id.nav_manage) {
        } else if (id == R.id.nav_share) {
        } else if (id == R.id.nav_send) {
        }

//        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
//        drawer.closeDrawer(GravityCompat.START);
        return true
    }

    override fun onClick(v: View) {}
    fun hideLayout() {
        tabs!!.visibility = View.GONE
        viewPager!!.visibility = View.GONE
    }

    fun showLayout() {
        tabs!!.visibility = View.VISIBLE
        viewPager!!.visibility = View.VISIBLE
    }

    fun LoadFragSearch(query: String?) {
        val bundle = Bundle()
        bundle.putString("query", query)
        val frag: Fragment = FragmentSearchNews()
        frag.arguments = bundle
        val manager = supportFragmentManager
        val transaction = manager.beginTransaction()
        hideLayout()
        transaction.replace(R.id.mainFrame, frag, "Search Fragment").addToBackStack(null)
        transaction.commit()
    }

    override fun onFragmentInteraction(uri: Uri?) {
    }
}